src/main/java/io/cos/cas/osf/web/flow/login/OsfUnsupportedInstitutionLoginPreparationAction.java

```diff
@@ -0,0 +1,39 @@
+package io.cos.cas.osf.web.flow.login;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.apereo.cas.authentication.adaptive.AdaptiveAuthenticationPolicy;
+import org.apereo.cas.web.flow.resolver.CasDelegatingWebflowEventResolver;
+import org.apereo.cas.web.flow.resolver.CasWebflowEventResolver;
+
+import org.springframework.webflow.execution.Event;
+import org.springframework.webflow.execution.RequestContext;
+
+/**
+ * This is {@link OsfUnsupportedInstitutionLoginPreparationAction}.
+ *
+ * Extends {@link OsfAbstractLoginPreparationAction} to prepare the unsupported institution login page.
+ *
+ * @author Longze Chen
+ * @since 20.1.0
+ */
+@Slf4j
+public class OsfUnsupportedInstitutionLoginPreparationAction extends OsfAbstractLoginPreparationAction {
```

Comment:
I don't think we need an "empty" action. The flow can jump directly to the new view / template. See the other comment.